tests/tt_eager/python_api_testing/unit_testing/misc/test_all_gather.py

```diff
@@ -91,27 +107,119 @@ def run_all_gather_on_t3000_impl(
     for i, t in enumerate(input_tensors):
         tt_input_tensors.append(ttl.tensor.Tensor(t, input_dtype).to(layout).to(devices[i], mem_config))
 
-    tt_out_tensors = ttl.tensor.all_gather(tt_input_tensors, dim, num_links, output_mem_config=mem_config)
+    for i in range(num_iters):
+        tt_out_tensors = ttl.tensor.all_gather(tt_input_tensors, dim, num_links, output_mem_config=mem_config)
+
+        for d in devices:
+            ttl.device.Synchronize(d)
+        print(f"Done iteration {i}")
 
     for i, t in enumerate(tt_out_tensors):
         tt_output_tensor = t.cpu().to(ttl.tensor.Layout.ROW_MAJOR).to_torch()
         if input_dtype == ttl.tensor.DataType.BFLOAT16:
             eq, output = comp_equal(tt_output_tensor, input_tensor)
         else:
             eq, output = comp_pcc(tt_output_tensor, input_tensor)
+        count = 0
+        if not eq:
+            print(f"output mismatch for tensor {i}")
```

Comment:
I only put this here for usability during sweeps. Sometimes during a large sweep a test will fail and it will just say `FAILED` so I don't know the type of error until the end.

Reply:
I see, can you switch to loguru then if we want to keep it?